tracee-rules/signatures/signaturestest/signaturestest.go

```diff
@@ -1,32 +1,35 @@
 package signaturestest
 
-import "github.com/aquasecurity/tracee/tracee-rules/types"
+import (
+	"github.com/aquasecurity/tracee/tracee-rules/types"
+)
 
-// SigTest is a utility to test signatures
-type SigTest struct {
-	// Events are a series of events that the signature will be tested with (in the provided order)
-	Events []types.Event
-	// CB provide a callback for this test case only if you want to check the Finding result
-	// if providing your own callback, make sure to call SetStatus from it
-	CB types.SignatureHandler
-	// Expect declares if we expect the signature to match for this event
-	Expect bool
-	// Status shows if the signatures has matched
-	Status bool
+// FindingsHolder is a utility struct that defines types.SignatureHandler
+// callback method and holds types.Finding values received as the callback's
+// argument in memory.
+type FindingsHolder struct {
+	Values []types.Finding
 }
 
-func (st *SigTest) Init(sig types.Signature) {
-	st.Status = false
-	if st.CB != nil {
-		sig.Init(func(res types.Finding) {
-			st.SetStatus(res)
-			st.CB(res)
-		})
-	} else {
-		sig.Init(st.SetStatus)
+func (h *FindingsHolder) OnFinding(f types.Finding) {
+	h.Values = append(h.Values, f)
+}
+
+func (h *FindingsHolder) GroupBySigID() map[string]types.Finding {
+	r := make(map[string]types.Finding)
+	for _, v := range h.Values {
+		r[v.SigMetadata.ID] = v
+	}
+	return r
+}
+
+func (h *FindingsHolder) FirstValue() *types.Finding {
+	if len(h.Values) == 0 {
+		return nil
 	}
+	return &h.Values[0]
 }
 
-func (st *SigTest) SetStatus(res types.Finding) {
-	st.Status = true
+func (h *FindingsHolder) Empty() bool {
+	return len(h.Values) == 0
```

Comment:
ditto

Reply:
Actually the Empty method is not used anywhere so I'm removing it.